### PR TITLE
Allow `nested_type` terraform attributes

### LINKED ISSUE #93: KeyError: 'type' with valid Ansible and Terrform data
##### SUMMARY
When running a `cloud.terraform.terraform` task to provision a Red Hat OpenShift Service on AWS (ROSA) instance, I am getting a `KeyError: 'type'`.
Both the Ansible and Terraform config are valid. If I go to the `project_path` and run `terraform plan` it succeeds.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
module_utils/models.py

https://github.com/ansible-collections/cloud.terraform/blob/main/plugins/module_utils/models.py

##### ANSIBLE VERSION
```bash
$ ansible --version
ansible [core 2.15.5]
  config file = /home/local-admin/Documents/code/github/aap-rosa-deploy/ansible.cfg
  configured module search path = ['/home/local-admin/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.11/site-packages/ansible
  ansible collection location = /home/local-admin/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.11.5 (main, Aug 28 2023, 00:00:00) [GCC 13.2.1 20230728 (Red Hat 13.2.1-1)] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

##### COLLECTION VERSION
```bash
$ ansible-galaxy collection list cloud.terraform

# /home/local-admin/.ansible/collections/ansible_collections
Collection      Version
--------------- -------
cloud.terraform 1.1.1  
```

##### CONFIGURATION
```bash
CONFIG_FILE() = /home/local-admin/Documents/code/github/aap-rosa-deploy/ansible.cfg
DEFAULT_HOST_LIST(/home/local-admin/Documents/code/github/aap-rosa-deploy/ansible.cfg) = ['/home/local-admin/Documents/code/github/aap-rosa-deploy/inventory.yaml']
EDITOR(env: EDITOR) = /usr/bin/vim
```

##### OS / ENVIRONMENT
```bash
$ cat /etc/redhat-release 
Fedora release 38 (Thirty Eight)
```

##### STEPS TO REPRODUCE
Apply the Terraform from a `project_path`.
The 

```yaml
# example task
- name: "Create ROSA cluster"
  environment:
    TF_BACKEND_BUCKET: "{{ rosa_cluster_name }}-terraform"
    TF_LOG: debug
    TF_LOG_PATH: "{{ work_dir }}/tf_rosa_deploy.log"
  cloud.terraform.terraform:
    complex_vars: false
    force_init: true
    project_path: "{{ work_dir }}/{{ rosa_cluster_name }}/terraform"
    provider_upgrade: true
    state: present
    variables:
      token: "{{ rosa_token }}"
      AWS_ACCESS_KEY_ID: "{{ aws_access_key_id }}"
      AWS_SECRET_ACCESS_KEY: "{{ aws_secret_access_key }}"
      AWS_DEFAULT_REGION: "{{ rosa_region }}"
  register: "create"
```

##### EXPECTED RESULTS
The module executes (whether the Terraform apply works or not).

##### ACTUAL RESULTS
```bash
TASK [rh.rosa.create : Create ROSA cluster: hub] *******************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'type'
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/home/local-admin/.ansible/tmp/ansible-tmp-1697092022.8988516-114662-223120601896804/AnsiballZ_terraform.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/local-admin/.ansible/tmp/ansible-tmp-1697092022.8988516-114662-223120601896804/AnsiballZ_terraform.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/local-admin/.ansible/tmp/ansible-tmp-1697092022.8988516-114662-223120601896804/AnsiballZ_terraform.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.cloud.terraform.plugins.modules.terraform', init_globals=dict(_module_fqn='ansible_collections.cloud.terraform.plugins.modules.terraform', _modlib_path=modlib_path),\n  File \"<frozen runpy>\", line 226, in run_module\n  File \"<frozen runpy>\", line 98, in _run_module_code\n  File \"<frozen runpy>\", line 88, in _run_code\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/modules/terraform.py\", line 659, in <module>\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/modules/terraform.py\", line 514, in main\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/module_utils/terraform_commands.py\", line 173, in providers_schema\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/module_utils/models.py\", line 202, in from_json\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/module_utils/models.py\", line 203, in <dictcomp>\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/module_utils/models.py\", line 186, in from_json\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/module_utils/models.py\", line 187, in <dictcomp>\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/module_utils/models.py\", line 157, in from_json\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/module_utils/models.py\", line 158, in <dictcomp>\n  File \"/tmp/ansible_cloud.terraform.terraform_payload_ub98fnq0/ansible_cloud.terraform.terraform_payload.zip/ansible_collections/cloud/terraform/plugins/module_utils/models.py\", line 128, in from_json\nKeyError: 'type'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

### PR DESCRIPTION
**Pull request by justinc1**
##### SUMMARY

`terraform providers schema -json` attributes can be nested. In this case, there is no `type` field, but we have `nested_type`.

The collection uses schema to sanitize `diff` in `terraform` module output.
If attribute is sensitive, its value will be hidden.

In case of nested attribute it might be possible to have top level attribute with `sensitive=False` containing a child attribute with `sensitive=True`. But I was not able to find terraform provider with such schema. I looked only at 35 official providers, and ignored remaining ~3000 partner and community providers.

The only official provider with `nested_type` attributes and also `sensitive=True` child was `waypoint`. A schema fragment is included into unit tests. `git_auth_basic` has non-sensitive `username` and sensitive `password`. Also `git_auth_basic` is marked as sensitive. Thus it seems it is not needed to modify logic for diff sanitization - whole `git_auth_basic` will be hidden.

Fixes #93

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

cloud.terraform.terraform

##### ADDITIONAL INFORMATION

Terraform provider `awscc` schema also contains nested attributes, an integration test is added to check schema is parsed.

EDIT:
```
terraform providers schema -json | jq '.provider_schemas["registry.terraform.io/terraform-redhat/rhcs"].resource_schemas.rhcs_cluster_rosa_classic.block.attributes.admin_credentials'
```
This is example of nested_type attribute that is not marked as sensitive, but it does contain sensitive sub-attribute.
Maybe a follow up PR will be need to ensure `admin_credentials.password` is hidden in ansible diff output.

### CODE REVIEW

File: plugins/modules/terraform.py [hunk 1]
```diff
@@ -288,7 +288,7 @@
 import dataclasses
 import os
 import tempfile
-from typing import List, Tuple
+from typing import List
```
Comment (abikouo) [inline plugins/modules/terraform.py:291]: this has also been resolved in the main branch

File: tests/integration/targets/awscc/meta/main.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+---
```
Comment (abikouo) [inline tests/integration/targets/awscc/meta/main.yml:1]: could you please delete this file?

**Comment by abikouo** [comment]
@justinc1 you need to rebase the PR to fix linters issue (especially the `isort` one on `tests/unit/plugins/modules/test_terraform.py`)

**Review by abikouo — APPROVED** [review]
LGTM! Thanks @justinc1

**Review by alinabuzachis — APPROVED** [review]
@justinc1 Thank you for working on this! LGTM!

**Pull request closed without merging**